Documentation/Configuration/Yaml/YamlApi.rst

```diff
@@ -73,75 +85,85 @@ Example:
 
-   $GLOBALS['TYPO3_CONF_VARS']['SYS']['yamlLoader']['placeholderProcessors'][\Vendor\MyExtension\PlaceholderProcessor\CustomPlaceholderProcessor::class] = [
-      'before' => [
-         \TYPO3\CMS\Core\Configuration\Processor\Placeholder\ValueFromReferenceArrayProcessor::class
-      ],
-      'after' => [
-         \TYPO3\CMS\Core\Configuration\Processor\Placeholder\EnvVariableProcessor::class
-      ],
-      'disabled' => false,
-   ];
+    $GLOBALS['TYPO3_CONF_VARS']['SYS']['yamlLoader']['placeholderProcessors']
+        [\Vendor\MyExtension\PlaceholderProcessor\CustomPlaceholderProcessor::class] = [
+            'before' => [
+                \TYPO3\CMS\Core\Configuration\Processor\Placeholder\ValueFromReferenceArrayProcessor::class
+            ],
+            'after' => [
+                \TYPO3\CMS\Core\Configuration\Processor\Placeholder\EnvVariableProcessor::class
+            ],
+            'disabled' => false,
+        ];
 
-New placeholder processors must implement the :php:`\TYPO3\CMS\Core\Configuration\Processor\Placeholder\PlaceholderProcessorInterface`
+New placeholder processors must implement the
+:php:`\TYPO3\CMS\Core\Configuration\Processor\Placeholder\PlaceholderProcessorInterface`.
 An implementation may look like the following:
 
-.. code-block:: php
-
-   class ExamplePlaceholderProcessor implements PlaceholderProcessorInterface
-   {
-      public function canProcess(string $placeholder, array $referenceArray): bool
-      {
-         return strpos($placeholder, '%example(') !== false;
-      }
-
-      public function process(string $value, array $referenceArray)
-      {
-         // do some processing
-         $result = $this->getValue($value);
-
-         // Throw this exception if the placeholder can't be substituted
-         if ($result === null) {
-            throw new \UnexpectedValueException('Value not found', 1581596096);
-         }
-         return $result;
-      }
-
-      private function getValue(string $value): ?string
-      {
-         // implement logic to fetch specific values from an external service
-         // or just add simple mapping logic - whatever is appropriate
-         $aliases = [
-            'foo' => 'F-O-O',
-            'bar' => 'ARRRRR',
-         ];
-         return $aliases[$value] ?? null;
-      }
-   }
-
-
-This may be used for example in the site configuration:
-
-.. code-block:: yaml
-
-   someVariable: '%example(somevalue)%'
-   anotherVariable: 'inline::%example(anotherValue)%::placeholder'
-
-If a new processor returns a string or number, it may also be used inline as above.
-If it returns an array, it cannot be used inline since the whole content will be replaced with the new value.
+..  code-block:: php
+    :caption: EXT:my_extension/Classes/Configuration/Processor/Placeholder/ExamplePlaceholderProcessor.php
+
+    final class ExamplePlaceholderProcessor implements PlaceholderProcessorInterface
```

Comment:
I would suggest to move this snippet into a separate php file that can be linted and have cgl applied to. then use the literalinclude directive